File: web/source/hypertext-transfer/http-methods.js
"use strict";

// @ts-check

export {
    HTTP_CONNECT,
    HTTP_DELETE,
    HTTP_GET,
    HTTP_HEAD,
    HTTP_OPTIONS,
    HTTP_PATCH,
    HTTP_POST,
    HTTP_PUT,
    HTTP_TRACE,
    httpMethods
}

// Hypertext Transfer Protocol Methods
/** @type {string} */
const HTTP_CONNECT = String("CONNECT");
/** @type {string} */
const HTTP_DELETE = String("DELETE");
/** @type {string} */
const HTTP_GET = String("GET");
/** @type {string} */
const HTTP_HEAD = String("HEAD");
/** @type {string} */
const HTTP_OPTIONS = String("OPTIONS");
/** @type {string} */
const HTTP_PATCH = String("PATCH");
/** @type {string} */
const HTTP_POST = String("POST");
/** @type {string} */
const HTTP_PUT = String("PUT");
/** @type {string} */
const HTTP_TRACE = String("TRACE");

// Hypertext Transfer Protocol Method Vector
/** @type {function(): string[][]} */
function httpMethods() {
    const hypertextTransferMethods = new Array([
        HTTP_CONNECT,
        HTTP_DELETE,
        HTTP_GET,
        HTTP_HEAD,
        HTTP_OPTIONS,
        HTTP_PATCH,
        HTTP_POST,
        HTTP_PUT,
        HTTP_TRACE
    ]);

    return hypertextTransferMethods;
}

File: src/hypertext_transfer/http_methods.rs
#![allow(dead_code)]

use std::{primitive::str, vec::Vec};

// Hypertext Transfer Protocol Method Defintion
pub type HTTPMethod = &'static str;

// Hypertext Transfer Protocol Methods
pub const HTTP_CONNECT: HTTPMethod = "CONNECT";
pub const HTTP_DELETE: HTTPMethod = "DELETE";
pub const HTTP_GET: HTTPMethod = "GET";
pub const HTTP_HEAD: HTTPMethod = "HEAD";
pub const HTTP_OPTIONS: HTTPMethod = "OPTIONS";
pub const HTTP_PATCH: HTTPMethod = "PATCH";
pub const HTTP_POST: HTTPMethod = "POST";
pub const HTTP_PUT: HTTPMethod = "PUT";
pub const HTTP_TRACE: HTTPMethod = "TRACE";

// Hypertext Transfer Protocol Method Vector
pub fn methods_vector() -> Vec<HTTPMethod> {
    let http_methods: Vec<HTTPMethod> = Vec::from([
        HTTP_CONNECT,
        HTTP_DELETE,
        HTTP_GET,
        HTTP_HEAD,
        HTTP_OPTIONS,
        HTTP_PATCH,
        HTTP_POST,
        HTTP_PUT,
        HTTP_TRACE,
    ]);

    return http_methods;
}
